projects/scripts/fmc_constr_generator.tcl

```diff
@@ -0,0 +1,276 @@
+proc gen_fmc_constr {{fmc_index1 fmc} {fmc_index2 {}}} {
+  
+  set project_name [get_property NAME [current_project]]
```

Comment:
Typically when we add support for a new project we do not start from an opened Vivado project.
If you see a value in using this script in an opened project, I suggest adding it as an extra feature not a base or a starting point.

The name of the carrier and daughter boards can be extracted from the current working directory.
set cwd [pwd]
set carrier [file tail $cwd]
set eval_board [file tail [file dirname $cwd]]

Also, in the current format of the code the script will fail.
e.g.: no files matched glob pattern "../../common/evb_zc706/evb_zc706_fmc_hpc*.txt"